### DISCUSSION TITLE
How is lsp being loaded with the lazyfile event?

### DISCUSSION THREAD

**Discussion by Me1m0th** [LazyVim/LazyVim#1953] [General]
I have copied the 'lazyfile' event from this repository but when I use it with lspconfig i have to :LspStart, if i started with an empty buffer at startup; after some searching I found this issue—which suggests doing something like: do filetype–: https://github.com/williamboman/mason.nvim/issues/126#issuecomment-1193651563
So, my question is, how is lazyvim attaching lsp to new buffers?

**Comment by dpetka2001** [comment]
From my understanding `LazyFile` is for the events `{ "BufReadPost", "BufNewFile", "BufWritePre" }`, it's just that the loading gets deferred to a bit later time, so that the UI renders first without it being blocked and thus being responsive for the user as soon as possible. But that only happens when you invoke Neovim with a file argument like `nvim init.lua`.

**Comment by Me1m0th** [comment]
I am aware of that; but how is lsp loaded when nvim starts with an empty buffer? Is some function being called to start lsp?--in my configuration lsp is loaded only when, as you have mentioned, you directly open a file, like: `nvim init.lua`. Sorry if it is a convoluted question.

**Reply by dpetka2001** [reply]
When you invoke Neovim with `nvim` and then open a buffer from the dashboard, then `lspconfig` gets lazy-loaded on `{ "BufReadPost", "BufNewFile", "BufWritePre" }` events and that attaches the corresponding LSP server based on the filetype of the buffer.

**Reply by dpetka2001** [reply]
When you say empty buffer, you mean like creating a new file from the dashboard? In that case the buffer has no filetype attached to it yet, until you save it with a filename. Since `lspconfig` was already loaded, no LSP server is attached to the buffer until you either invoke `:e` to reload the buffer or press `:LspStart` for it to attach.

**Reply by Me1m0th** [reply]
Hmm... thank you very much; is there perhaps a way to automate having to do :LspStart/:e eachtime?

**Reply by dpetka2001** [reply]
No idea. Maybe try and not open an empty buffer? I don't really understand your workflow and also don't have that much knowledge concerning LSP servers.

**Reply by Me1m0th** [reply]
Lol. Thank you for all the help! Closing the discussion.
